app/services/BackgroundTaskService.js

```diff
@@ -0,0 +1,135 @@
+import BackgroundFetch from 'react-native-background-fetch';
+import PushNotification from 'react-native-push-notification';
+import { IntersectSet } from '../helpers/Intersect';
+import { GetStoreData, SetStoreData } from '../helpers/General';
+import { isPlatformiOS } from './../Util';
+import languages from '../locales/languages';
+
+const INTERSECT_INTERVAL = 30 / 60; // In Minutes
+
+export function executeTask() {
+  check_intersect = () => {
```

Comment:
I think `check_intersect` should be defined over in `Intersect.js`.  I think it's OK to keep the executeTask wrapper for the background service to call ... keeps it a bit cleaner by separating out what will execute from the service definition, which I think is good.